unit_tests/test_rest_asset_client.py

```diff
@@ -0,0 +1,297 @@
+# pylint: disable=missing-function-docstring, no-member, too-many-public-methods
+import uuid
+from unittest.mock import patch, MagicMock
+# import os
+
+import pytest
+from gsy_e_sdk.constants import MAX_WORKER_THREADS
+
+from gsy_e_sdk.clients.rest_asset_client import RestAssetClient
+
+TEST_ASSET_UUID = str(uuid.uuid4())
+TEST_SIMULATION_ID = str(uuid.uuid4())
+TEST_AGGREGATOR_UUID = str(uuid.uuid4())
+TEST_TRANSACTION_ID = str(uuid.uuid4())
+
+TEST_DOMAIN_NAME = "test.domain@name.com"
+TEST_WEBSOCKETS_DOMAIN_NAME = "wss:test.domain@name.com"
+
+TEST_COMMAND_RESPONSE = {"command": "register",
+                         "transaction_id": TEST_TRANSACTION_ID,
+                         "registered": True
+                         }
+
+
+@pytest.fixture(name="mock_connections", autouse=True)
+def fixture_mock_connections(mocker):
+    # gsy_framework/client_connections/utils.py
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.retrieve_jwt_key_from_server")
+    mocker.patch("gsy_framework.client_connections.utils.RepeatingTimer")
+
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.WebsocketThread")
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.ThreadPoolExecutor")
+
+
+@pytest.fixture(name="mock_environment_use_functions")
+def fixture_mock_env_use_functions(mocker):
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.simulation_id_from_env",
+                 return_value=TEST_SIMULATION_ID)
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.domain_name_from_env",
+                 return_value=TEST_DOMAIN_NAME)
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.websocket_domain_name_from_env",
+                 return_value=TEST_WEBSOCKETS_DOMAIN_NAME)
+
+
+@pytest.fixture(name="mock_transaction_id")
+def mock_transaction_id_and_timeout_blocking(mocker):
+    mocker.patch("gsy_framework.client_connections.utils.uuid.uuid4",
+                 return_value=TEST_TRANSACTION_ID)
+
+
+@pytest.fixture(name="client")
+def fixture_rest_asset_client(mock_environment_use_functions):  # pylint: disable=unused-argument
+    return RestAssetClient(asset_uuid=TEST_ASSET_UUID)
+
+
+@pytest.mark.usefixtures("mock_environment_use_functions")
+class TestRestAssetClient:
+    """Test methods for RestAssetClient class."""
+
+    @staticmethod
+    @pytest.mark.parametrize("set_value, expected_ret_val",
+                             [(TEST_SIMULATION_ID, TEST_SIMULATION_ID),
+                              (None, TEST_SIMULATION_ID)])
+    def test_constructor_simulation_id_setup(set_value, expected_ret_val):
```

Comment:
Nit, a docstring here could be useful (please feel free to rephrase it in your style, mine is just an example)
```suggestion
    def test_constructor_simulation_id_setup(set_value, expected_ret_val):
    """Test whether the simulation id will be set regardless of whether it was explicitly passed or not (should read the env variable)"""
```